Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java

```diff
@@ -494,6 +521,33 @@ protected void onActivityResult(int requestCode, int resultCode, Intent result)
         NativeLibrary.ChangeDisc(result.getData().toString());
       }
     }
+    else if (requestCode == REQUEST_SKYLANDER_FILE)
+    {
+      Pair<Integer, String> slot =
+              SkylanderConfig.loadSkylander(mSkylanderSlots.get(skylanderSlot).getPortalSlot(),
+                      result.getData().toString());
+      clearSkylander(skylanderSlot);
+      mSkylanderSlots.get(skylanderSlot).setPortalSlot(slot.first);
+      mSkylanderSlots.get(skylanderSlot).setLabel(slot.second);
+      mSkylandersBinding.skylandersManager.getAdapter().notifyItemChanged(skylanderSlot);
+      skylanderSlot = -1;
+      skylanderData = new Pair(new Pair<>(-1, -1), "Slot");
+    }
+    else if (requestCode == REQUEST_CREATE_SKYLANDER)
+    {
+      if (!(skylanderData.first.first == -1) && !(skylanderData.first.second == -1))
+      {
+        Pair<Integer, String> slot = SkylanderConfig.createSkylander(skylanderData.first.first,
+                skylanderData.first.second,
+                result.getData().toString(), mSkylanderSlots.get(skylanderSlot).getPortalSlot());
+        clearSkylander(skylanderSlot);
+        mSkylanderSlots.get(skylanderSlot).setPortalSlot(slot.first);
+        mSkylanderSlots.get(skylanderSlot).setLabel(slot.second);
+        mSkylandersBinding.skylandersManager.getAdapter().notifyItemChanged(skylanderSlot);
+        skylanderSlot = -1;
+        skylanderData = new Pair(new Pair<>(-1, -1), "Slot");
+      }
+    }
```

Comment:
Would you mind changing this so that all of onActivityResult is surrounded with `if (resultCode == RESULT_OK)`? (except for the super call)
You need to for when you back out of loading a skylander file and it should just reduce redundant code.

Reply:
Can do